get-indirection-metrics/index.js

```diff
@@ -0,0 +1,107 @@
+const { Octokit } = require("@octokit/rest");
+const fs = require("fs");
+const dig = require("node-dig-dns");
+
+const octokit = new Octokit({
+  auth: process.env.GITHUB_TOKEN,
+});
+
+const INDIRECTION_REPO = "isomerpages/isomer-indirection";
+
+async function getAllIndirectionDomains() {
+  const dnsTree = await octokit.git.getTree({
+    owner: INDIRECTION_REPO.split("/")[0],
+    repo: INDIRECTION_REPO.split("/")[1],
+    tree_sha: "main",
+    recursive: 1,
+  });
+
+  const dnsFiles = dnsTree.data.tree.filter((file) =>
+    file.path.startsWith("dns/")
+  );
+
+  const domains = dnsFiles.map((file) => file.path.split("/")[1].slice(0, -3));
+
+  return domains;
+}
+
+async function getDomainCNAMERecord(domain) {
+  let result = await dig([domain, "CNAME"]);
+  if (!Object.keys(result).includes("answer") || result.answer.length === 0) {
+    result = await dig([`www.${domain}`, "CNAME"]);
+  }
+
+  return result;
+}
+
+async function getDomainsOnIndirectionLayer(domains) {
+  const domainsOnIndirectionLayer = [];
+  process.stdout.write(`Checking domain 0/${domains.length}...`);
+
+  for (let i = 0; i < domains.length; i++) {
+    const result = await getDomainCNAMERecord(domains[i]);
+    process.stdout.clearLine(0);
+    process.stdout.cursorTo(0);
```

Comment:
just curious, this clears the line and sets the cursor to the initial position right?

Reply:
Yup, also added a comment in ba907ffbc01dcb04a72ead1e268372e2e31720cf. Wanted to try something new here.